### ISSUE TITLE
Sub-URI issues with hardcoded URLs in JS

### ISSUE THREAD

**Issue by fbacall** [seek4science/seek#938]
Lots of places have root-relative paths hard-coded in the JS like this:
https://github.com/seek4science/seek/blob/seek-1.12/app/assets/javascripts/project_folders.js#L108
https://github.com/seek4science/seek/blob/seek-1.12/app/assets/javascripts/openbis.js#L59 (Also this route is non-existent...)

this won't work if the app is deployed under a sub-URI. Need to replace these with something like this:
https://github.com/seek4science/seek/blob/seek-1.12/app/assets/javascripts/sharing.js.erb#L79

If there are URL parameters required (IDs etc.) these can be added using placeholders that are dynamically substituted at run-time in the JS, e.g.
```
var showDatasetFilesPath = '<%= Rails.application.routes.url_helpers.show_dataset_files_openbis_endpoint_openbis_dataset_path('-endpoint_id-', '-dataset_id-') %>';

...

var path = showDatasetFilesPath.replace('-endpoint_id-', endpointId).replace('-dataset_id-', permId);
```

**Comment by stuzart** [comment]
can you `include UrlHelper` that makes it a bit cleaner and just call `show_dataset_files_openbis_endpoint_openbis_dataset_path` in the example above?

**Comment by fbacall** [comment]
I've tried to avoid including routes as much as possible, because I've had obscure errors where it clobbers some Rails methods and causes strange behaviour. It's probably OK to include in JavaScript though, and is done in other files.

**Comment by fbacall** [comment]
Fixed by #946
